File: skillful_network_client/src/app/shared/models/training.ts
import{ITraining} from './mock.training'
export class Training {
    // tslint:disable-next-line:variable-name
    private _id: number;
    // tslint:disable-next-line:variable-name
    private _name: string;
    private _organisation: string;
    private _description: string;
    private _financer: string;
    private _dateBeg: Date;
    private _dateEnd: Date;
    private _durationHours: number;
    private _dateUpload: Date;
    private _prerequisites: string[];
    private _keywords: string[];

    constructor(data: ITraining) {
        this.id = data.id;
        this.name = data.name;
        this.organisation = data.organisation;
        this.description = data.description;
        this.financer = data.financer;
        this.dateBeg = data.dateBeg;
        this.dateEnd = data.dateEnd;
        this.durationHours = data.durationHours;
        this.dateUpload = data.dateUpload;
        this.prerequisites = data.prerequisites;
        this.keywords = data.keywords;
    }

    /* GETTERS & SETTERS */
    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get organisation(): string {
        return this._organisation;
    }

    set organisation(value: string) {
        this._organisation = value;
    }
    get description(): string {
        return this._description;
    }

    set description(value: string) {
        this._description = value;
    }

    get keywords(): string[] {
        return this._keywords;
    }

    set keywords(value: string[]) {
        this._keywords = value;
    }

    get prerequisites(): string[] {
        return this._prerequisites;
    }

    set prerequisites(value: string[]) {
        this._prerequisites = value;
    }

    get dateUpload(): Date {
        return this._dateUpload;
    }

    set dateUpload(value: Date) {
        this._dateUpload = value;
    }

    get durationHours(): number {
        return this._durationHours;
    }

    set durationHours(value: number) {
        this._durationHours = value;
    }

    get dateEnd(): Date {
        return this._dateEnd;
    }

    set dateEnd(value: Date) {
        this._dateEnd = value;
    }

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    get dateBeg(): Date {
        return this._dateBeg;
    }

    set dateBeg(value: Date) {
        this._dateBeg = value;
    }

    get financer(): string {
        return this._financer;
    }

    set financer(value: string) {
        this._financer = value;
    }
}

File: skillful_network_server/src/main/java/fr/uca/cdr/skillful_network/model/entities/Training.java
package fr.uca.cdr.skillful_network.model.entities;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import fr.uca.cdr.skillful_network.model.entities.simulation.exercise.Keyword;

import java.util.Arrays;
import java.util.Date;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

@Entity
@Table(name = "training")
public class Training {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private String name;
	private String organisation;
	private String description;
	private String financer;
	private Date dateBeg;
	private Date dateEnd;
	private Date dateUpload;
	private Long durationHours;
	private String[] prerequisites;
	@ManyToMany(fetch = FetchType.EAGER, cascade =  CascadeType.PERSIST)
	private Set<Keyword> keywords = new HashSet<>();

	@OneToMany(mappedBy = "training", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
	private Set<TrainingApplication> trainingApplicationSet = new HashSet<>();

	@ManyToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "user_id")
	@JsonIgnoreProperties("trainingSet")
	private User user;

	public Set<TrainingApplication> getTrainingApplicationSet() {
		return trainingApplicationSet;
	}

	public void setTrainingApplicationSet(Set<TrainingApplication> trainingApplicationSet) {
		this.trainingApplicationSet = trainingApplicationSet;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public String getOrganisation() {
		return organisation;
	}

	public void setOrganisation(String organisation) {
		this.organisation = organisation;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getFinancer() {
		return financer;
	}

	public void setFinancer(String financer) {
		this.financer = financer;
	}

	public Date getDateBeg() {
		return dateBeg;
	}

	public void setDateBeg(Date dateBeg) {
		this.dateBeg = dateBeg;
	}

	public Date getDateEnd() {
		return dateEnd;
	}

	public void setDateEnd(Date dateEnd) {
		this.dateEnd = dateEnd;
	}

	public Date getDateUpload() {
		return dateUpload;
	}

	public void setDateUpload(Date dateUpload) {
		this.dateUpload = dateUpload;
	}

	public Long getDurationHours() {
		return durationHours;
	}

	public void setDurationHours(Long durationHours) {
		this.durationHours = durationHours;
	}

	public String[] getPrerequisites() {
		return prerequisites;
	}

	public void setPrerequisites(String[] prerequisites) {
		this.prerequisites = prerequisites;
	}

	public Set<Keyword> getKeywords() {
		return keywords;
	}

	public void setKeywords(Set<Keyword> keywords) {
		this.keywords = keywords;
	}

	public Training() {
		super();
	}

	public Training(Long id, String name, String organisation, String description, String financer, Date dateBeg,
			Date dateEnd, Date dateUpload, Long durationHours, String[] prerequisites, Set<Keyword> keywords) {
		super();
		this.id = id;
		this.name = name;
		this.organisation = organisation;
		this.description = description;
		this.financer = financer;
		this.dateBeg = dateBeg;
		this.dateEnd = dateEnd;
		this.dateUpload = dateUpload;
		this.durationHours = durationHours;
		this.prerequisites = prerequisites;
		this.keywords = keywords;
	}

	public Training(String name, String organisation, String description, String financer, Date dateBeg, Date dateEnd,
			Date dateUpload, Long durationHours, String[] prerequisites, Set<Keyword> keywords) {
		super();
		this.name = name;
		this.organisation = organisation;
		this.description = description;
		this.financer = financer;
		this.dateBeg = dateBeg;
		this.dateEnd = dateEnd;
		this.dateUpload = dateUpload;
		this.durationHours = durationHours;
		this.prerequisites = prerequisites;
		this.keywords = keywords;
	}

	public Training(String name, String organisation, String description, String financer, Date dateBeg, Date dateEnd,
			Date dateUpload, Long durationHours) {
		super();
		this.name = name;
		this.organisation = organisation;
		this.description = description;
		this.financer = financer;
		this.dateBeg = dateBeg;
		this.dateEnd = dateEnd;
		this.dateUpload = dateUpload;
		this.durationHours = durationHours;
	}

	@Override
	public String toString() {
		return "Training [id=" + id + ", name=" + name + ", organisation=" + organisation + ", description="
				+ description + ", financer=" + financer + ", dateBeg=" + dateBeg + ", dateEnd=" + dateEnd
				+ ", dateUpload=" + dateUpload + ", durationHours=" + durationHours + ", prerequisites="
				+ Arrays.toString(prerequisites) + ", keywords=" + keywords + "]";
	}

	@Override
	public int hashCode() {
		return Objects.hash(id, name, organisation);
	}

}
